src/Workspaces/CSharp/Portable/Classification/Worker.cs

```diff
@@ -262,5 +265,52 @@ private void ClassifyDisabledText(SyntaxTrivia trivia, SyntaxTriviaList triviaLi
                 AddClassification(trivia, ClassificationTypeNames.ExcludedCode);
             }
         }
+
+        private void ClassifyLiteralAndSuffix(TextSpan tokenSpan, string literalClassification, int suffixLength)
+        {
+            var split = tokenSpan.End - suffixLength;
+            AddClassification(TextSpan.FromBounds(tokenSpan.Start, split), literalClassification);
+            AddClassification(TextSpan.FromBounds(split, tokenSpan.End), ClassificationTypeNames.Keyword);
+        }
+
+        private static bool TryGetNumericLiteralSuffix(string tokenText, out int suffixLength)
```

Comment:
We should refactor the code to share the existing logic here:
https://github.com/dotnet/roslyn/blob/0629a8d8fce24533aa27be798861b4cc316f5cf8/src/Features/Core/Portable/ConvertNumericLiteral/AbstractConvertNumericLiteralCodeRefactoringProvider.cs#L130-L135

@CyrusNajmabadi for suggestions on where to place the resulting shared method

Reply:
@jhinder Note that this is _not_ a request to switch to the regex form. The regex use would be inefficient compared to directly coded. After the change, neither location would be using regex for this.